### DISCUSSION TITLE
ADC resolution problem

### DISCUSSION THREAD

**Discussion by Semsem-220109** [stm32duino/Arduino_Core_STM32#1989] [Q&A] [answered]
Hi everyone,
When I use the analogRead() function, it always give me a value going from 0 to 1023 (10-bit) not 12-bit (0-4095) on the STM32F103C8T6 as it is capable of an ADC resolution of 12-bit
How to solve this ?

**Comment by fpistm** [comment, marked as answer]
Use analogReadResolution API:
https://www.arduino.cc/reference/en/language/functions/zero-due-mkr-family/analogreadresolution/

**Reply by Semsem-220109** [reply]
Isn't the STM32F103C8T6 meant to automatically read analog values with a 12-bit resolution ?

**Reply by fpistm** [reply]
this core follows Arduino API reference. `analogRead()` default resolution is 10bits anyway the acquisition is done with max resolution then mapped to 10bits.
https://www.arduino.cc/reference/en/language/functions/analog-io/analogread/
